proposals/umoci.md

```diff
@@ -0,0 +1,113 @@
+ * Experimental design work for a new OCI image-spec layer format (code-named "OCIv2").
+
+[libpathrs]: https://github.com/openSUSE/libpathrs
+
+## Frequently Asked Questions (FAQ)
+> *Does this proposal change the OCI Charter?*
+
+This proposal does not intend to amend the [OCI Charter][oci-charter].
+
+[oci-charter]: https://github.com/opencontainers/tob/blob/master/CHARTER.md
+
+> *Where does umoci fit into the OCI suite of projects?*
+
+umoci is intended to be a reference implementation of the OCI image-spec, and to fill a role akin to runc's relationship with the OCI runtime-spec.
+While umoci is usable on its own, it is expected that the vast majority of umoci's users will be consuming it as part of a larger build system which uses umoci internally to abstract away the specifics of the OCI image-spec.
+
+> *What about image-tools?*
+
+The original intention of umoci was to be a wrapper around the OCI image-tools and slowly upstream the work.
+Unfortunately, in the past 4 years, the image-tools project has not developed at a consistent pace -- and umoci works today and is fairly widely used.
+It is far more important that the OCI provide a solid "works out of the box" image manipulation tool, and umoci is (in our view) the best candidate for this role.
+
+In a future proposal, we may decide to sub-tree merge the OCI image-tools project into umoci (and archive the original project) in order to maintain the legacy of the work done by previous contributors and maintainers.
+Alternatively we may decide to split out the validation code from the OCI image-tools project and place it inside an OCI conformance project.
+However, we believe that these decisions should be considered a separate topic to the question of umoci's inclusion in the OCI.
+
+> *Why bless umoci over other alternative tools?*
+
+umoci is the only OCI image tool which was designed to be usable as a component to build other image-building tools.
+This has proven to be a practical benefit, with several tools ([KIWI][kiwi], [stacker][stacker], and several others) being built on top of umoci.
+In addition, umoci has been used by the [LXC project][lxc-oci] to support OCI-based images from *outside* the OCI ecosystem, something that was only reasonable because of umoci's unopinionated stance on images.
+umoci has seen production usage for several years as part of the [Open Build Project][obs]'s build system, building container images using [kiwi][kiwi].
+
+[kiwi]: https://osinside.github.io/kiwi/building/build_docker_container.html
+[stacker]: https://github.com/anuvu/stacker
+[lxc-oci]: https://github.com/lxc/lxc/blob/lxc-4.0.2/templates/lxc-oci.in
+[obs]: https://openbuildservice.org/
+
+> *How do we avoid the runc issue with implementation-specific quirks becoming a de-facto standard?*
+
+When developing new features in umoci (which are within the scope of the image-spec), a strong effort will be made to include those features in the upstream specification.
```

Comment:
got all the way through the proposal.. and didn't see a list of features in umoci, what it actually does, just abstract descriptions of a tool used by other image tools... and tool for working with OCI images that replaces at least a portion of OCI image-tools with a more user oriented cli. Some of the following may help in the understanding of what it does:

```
NAME:
   umoci - umoci modifies Open Container images

USAGE:
   umoci [global options] command [command options] [arguments...]

VERSION:
   0.3.1

AUTHOR(S):
   Aleksa Sarai <asarai@suse.com>

COMMANDS:
     raw      advanced internal image tooling
     help, h  Shows a list of commands or help for one command

   image:
     config      modifies the image configuration of an OCI image
     unpack      unpacks a reference into an OCI runtime bundle
     repack      repacks an OCI runtime bundle into a reference
     new         creates a blank tagged OCI image
     tag         creates a new tag in an OCI image
     remove, rm  removes a tag from an OCI image
     stat        displays status information of an image manifest

   layout:
     gc        garbage-collects an OCI image's blobs
     init      create a new OCI layout
     list, ls  lists the set of tags in an OCI image
```

Reply:
Sure, I can give a more concrete explanation of what it does.